src/arkode/arkode_io.c

```diff
@@ -1819,16 +2609,24 @@ int arkGetNumConstrFails(void* arkode_mem, long int* nconstrfails)
   }
   ark_mem = (ARKodeMem)arkode_mem;
 
+  /* Guard against use for non-adaptive time stepper modules */
+  if (!ark_mem->step_supports_adaptive)
+  {
+    arkProcessError(ark_mem, ARK_STEPPER_UNSUPPORTED, __LINE__, __func__,
+                    __FILE__, "time-stepping module does not support temporal adaptivity");
+    return (ARK_STEPPER_UNSUPPORTED);
+  }
+
   *nconstrfails = ark_mem->nconstrfails;
   return (ARK_SUCCESS);
 }
 
 /*---------------------------------------------------------------
-  arkGetNumExpSteps:
+  ARKodeGetNumExpSteps:
 
   Returns the current number of stability-limited steps
   ---------------------------------------------------------------*/
-int arkGetNumExpSteps(void* arkode_mem, long int* nsteps)
+int ARKodeGetNumExpSteps(void* arkode_mem, long int* nsteps)
```

Comment:
I was thinking that since non-adaptive steppers cannot report on either stability-limited *or* accuracy-limited steps at all, so "unsupported" felt more appropriate than 0, but I'm open to other interpretations.